src/components/Settings/SettingsAddressbookShare.vue

```diff
@@ -85,32 +86,88 @@ export default {
 	data() {
 		return {
 			isLoading: false,
-			usersOrGroups: [],
-			selectedUserOrGroup: []
+			inputGiven: false,
+			usersOrGroups: []
+		}
+	},
+	computed: {
+		placeholder() {
+			return t('contacts', 'Share with users or groups')
+		},
+		noResult() {
+			return t('contacts', 'No users or groups')
 		}
 	},
 	methods: {
-		limitText(count) {
-			return `and ${count} other users or groups`
+		/**
+		 * Share addressbook
+		 *
+		 * @param {Object} chosenUserOrGroup
+		 */
+		shareAddressbook({ sharee, id, group }) {
+			let addressbook = this.addressbook
+			this.$store.dispatch('shareAddressbook', { addressbook, sharee, id, group })
 		},
-
-		/* example :OC.linkToOCS('cloud', 2)+ 'groups?search=Test' */
+		/**
+		 * Format responses from axios.all and add them to the option array
+		 *
+		 * @param {Array} matches Array of matches returned from the axios request
+		 * @param {String} query
+		 * @param {Boolean} group
+		 */
+		formatMatchResults(matches, query, group) {
+			if (matches.length < 1) {
+				return
+			}
+			let regex = new RegExp(query, 'i')
+			let existingSharees = this.addressbook.shares.map(share => share.id + share.group)
+			matches = matches.filter(share => existingSharees.indexOf(share.id + group) === -1)
+			for (let i = 0; i < matches.length; i++) {
+				let matchResult = matches[i].displayname.split(regex)
+				let newMatch = {
+					sharee: matches[i].displayname,
+					id: matches[i].id,
+					matchstart: matchResult[0],
+					matchpattern: matches[i].displayname.match(regex)[0],
+					matchend: matchResult[1],
+					matchtag: group ? '(group)' : '(user)',
+					group: group
+				}
+				this.usersOrGroups.push(newMatch)
+			}
+		},
+		/**
+		 * Use Axios api call to find matches to the query from the existing Users & Groups
+		 *
+		 * @param {String} query
+		 */
 		asyncFind(query) {
 			this.isLoading = true
 			this.usersOrGroups = []
-			// let response = OC.linkToOCS('cloud', 2) + 'groups?search=' + query
-			fetch(OC.linkToOCS('cloud', 2) + 'groups?search=' + query).then(response => {
-				this.usersOrGroups.push(response)
-				this.isLoading = false
-			})
-			console.log(this.usersOrGroups) // eslint-disable-line
-			/* ajaxFindCountry(query).then(response => {
-				this.countries = response
-				this.isLoading = false
-			}) */
-		},
-		clearAll() {
-			this.selectedUserOrGroup = []
+			if (query.length > 0) {
+				api.all([
+					api.get(OC.linkToOCS('cloud', 2) + 'users/details?search=' + query),
+					api.get(OC.linkToOCS('cloud', 2) + 'groups/details?search=' + query)
+				]).then(response => {
+					let matchingUsers = Object.values(response[0].data.ocs.data.users)
+					let matchingGroups = response[1].data.ocs.data.groups
+					try {
+						debounce(this.formatMatchResults(matchingUsers, query, false), 250, true)
+					} catch (error) {
+						console.debug(error)
+					}
+					try {
+						debounce(this.formatMatchResults(matchingGroups, query, true), 250, true)
```

Comment:
This is wrongly positioned, the debounce should be before the request, or we'll still get a big amount of requests which data won't be up to date. :)

See https://github.com/nextcloud/contacts/blob/vue/src/components/properties/PropertyMultipleText.vue#L123

Reply:
I've taken 500 for the wait, correct?